lib/util/Queue.js

```diff
@@ -37,19 +45,28 @@ Queue.prototype.enqueue = function(id, thing) {
         item: thing,
         defer: defer
     });
-    // always process stuff asyncly, never syncly.
-    process.nextTick(() => {
-        this._consume();
-    });
+    if (!this._intervalMs) {
```

Comment:
For the same reason as https://github.com/matrix-org/matrix-appservice-irc/pull/141/files#r76762550 - you cannot rely on truthiness of integers if you aren't asserting for negative numbers.